salt/modules/esxcluster.py

```diff
@@ -22,5 +23,27 @@ def __virtual__():
     return (False, "Must be run on a proxy minion")
 
 
+def _deprecation_message(function):
```

Comment:
Non-blocking: some de-duping for this decorator would help, but I can't think of a place to put it 😂

Reply:
I stole from modules/azurearm_resources.py, used in a number of the other azurearm_xxxx.py files too, so followed that style. Liked the quick and easy style of one place to edit in a file, but yup, nice if there was a standard place for deprecation to go and them import, maybe a search for decorators to go in utils. But leave that for later.